### PR TITLE
Add files option for npm, fix indents on tests

### PR DESCRIPTION
**Pull request by vsn4ik**
1. Fix references in docs
1. Update change events in docs
1. Fix idents on tests
1. Move rows from `.npmignore` to `.gitignore`
1. Add `files` option on `package.json`
1. Use https when possible
1. ...

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -59,7 +59,6 @@ module.exports = function(grunt){
         },
         concat: {
             options: {
-                banner: '<%= banner %>',
```
Comment (acrobat) [inline Gruntfile.js]: why did you remove this line?

File: Gruntfile.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ module.exports = function(grunt){
             ' * Improvements by Andrew Rowls',
             ' * Licensed under the Apache License v2.0 (http://www.apache.org/licenses/LICENSE-2.0)',
             ' */'
-        ].join('\n'),
+        ].join('\n') + '\n',
```
Comment (acrobat) [inline Gruntfile.js]: Why is the extra new line needed?

**Comment by acrobat** [comment]
Thanks @vsn4ik for the cleanup! Can you restore these files:

- .npmignore (this was added in a pr for a specific case, so we should keep it for now (possible cleanup in 2.0)
- tests/assets/jquery-2.2.3.min.js (I've added this file yesterday to start some work on jquery 1 and 2 test suites)

File: Gruntfile.js (hunk 1 above)
Comment (vsn4ik) [inline Gruntfile.js]: For unification banner for js and css files.

File: Gruntfile.js (hunk 2 above)
Comment (vsn4ik) [inline Gruntfile.js]: For unification banner for js and css files.

**Comment by vsn4ik** [comment]
@acrobat I'm not going to restore unwanted files and unused libraries.

File: Gruntfile.js (hunk 1 above)
Comment (acrobat) [inline Gruntfile.js]: ah yes I see, it's now in the usebanner task!

**Comment by acrobat** [comment]
Ok, I will add the new jquery file in a separate PR for extra tests. But the .npmignore file was added to fix #1742, so we should atleast keep it until 2.0 (then we can remove it and fix things an other way)

I see you did a lot of small changes to the main js file, the changes look good on first sight. Can you guys also take a look @hebbet @Azaret, thanks!

**Comment by acrobat** [comment]
@vsn4ik thanks for fixing the comments! Is it maybe possible to move the changes to `bootstrap-datepicker.js` to a separate pull request. Other changes can stay in this PR and are ready for a merge!

**Comment by vsn4ik** [comment]
Done.

**Comment by acrobat** [comment]
Thanks @vsn4ik! Will review the other PR soon!
